Simplenote/AuthViewController+Swift.swift

```diff
@@ -172,12 +190,43 @@ extension AuthViewController {
                 self.showAuthenticationError(forCode: result.statusCode, responseString: nil)
             }
 
-            self.stopSignupAnimation()
+            self.stopActionAnimation()
             self.setInterfaceEnabled(true)
         }
     }
+    
+    @IBAction
+    func handleNewlineInField(_ field: NSControl) {
+        if field.isEqual(passwordField.textField) {
+            performMainAction(field)
+            return
+        }
+        
+        if field.isEqual(usernameField.textField), mode.isPasswordVisible == false {
+            performMainAction(field)
+        }
+    }
 }
 
+
+// MARK: - Animations
+//
+extension AuthViewController {
+    
+    @objc
+    func startActionAnimation() {
+        actionButton.title = mode.primaryActionAnimationText
+        actionProgress.startAnimation(nil)
+    }
+
+    @objc
+    func stopActionAnimation() {
+        actionButton.title = mode.primaryActionText
+        actionProgress.stopAnimation(nil)
+    }
```

Comment:
Rather than having N methods that set different strings, the text rendered while the spinner is onscreen is returned by the AuthenticationMode